### PR TITLE
Update impersonation_recipient_domain_display_name_subject.yml

### PR DESCRIPTION
**Pull request by morriscode**
Adding min length requirements to recipient and mailbox slds to prevent short domains < 4 chars from firing

### CODE REVIEW

File: detection-rules/impersonation_recipient_domain_display_name_subject.yml [hunk 1]
```diff
@@ -66,15 +70,14 @@ source: |
     or sender.email.domain.root_domain not in $high_trust_sender_root_domains
   )
   and (
-    (
-      not profile.by_sender().solicited
-    )
+    (not profile.by_sender().solicited)
```
Comment (jkamdjou) [inline detection-rules/impersonation_recipient_domain_display_name_subject.yml:73]:
```suggestion
    not profile.by_sender().solicited
```

File: detection-rules/impersonation_recipient_domain_display_name_subject.yml [hunk 2]
```diff
@@ -9,10 +9,13 @@ source: |
...
     (
-       // these are usually targeted with just 1 recipient,
+      // these are usually targeted with just 1 recipient,
       // but sometimes they CC themselves or have a blank CC
-      length(recipients.to) + length(recipients.cc) + length(recipients.bcc) <= 2
+      length(recipients.to) + length(recipients.cc)
+   + length(recipients.bcc) <= 2
       and any(recipients.to,
+              length(.email.domain.sld) >= 4
+              and
```
Comment (jkamdjou) [inline detection-rules/impersonation_recipient_domain_display_name_subject.yml:18]: weird formatting here

File: detection-rules/impersonation_recipient_domain_display_name_subject.yml [hunk 3]
```diff
@@ -9,10 +9,13 @@ source: |
   and (
     // recipient SLD is being impersonated in the subject + display name
     (
-       // these are usually targeted with just 1 recipient,
+      // these are usually targeted with just 1 recipient,
       // but sometimes they CC themselves or have a blank CC
-      length(recipients.to) + length(recipients.cc) + length(recipients.bcc) <= 2
+      length(recipients.to) + length(recipients.cc)
+   + length(recipients.bcc) <= 2
```
Comment (jkamdjou) [inline detection-rules/impersonation_recipient_domain_display_name_subject.yml:15]: weird formatting here, probably wanna wrap this math in parens anyway

File: detection-rules/impersonation_recipient_domain_display_name_subject.yml [hunk 4]
```diff
@@ -22,37 +25,38 @@ source: |
     or (
       // accounts for BCC'd messages where the recipients are empty
       // if BCC, sometimes the recipient will be the attacker's email
-      length(recipients.to) + length(recipients.cc) + length(recipients.bcc) <= 2
+      length(recipients.to) + length(recipients.cc)
+   + length(recipients.bcc) <= 2
```
Comment (jkamdjou) [inline detection-rules/impersonation_recipient_domain_display_name_subject.yml:29]: ditto here

**Review by aidenmitchell — APPROVED** [review]

**Pull request merged**